src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js

```diff
@@ -119,6 +123,7 @@ class CommentForm extends React.Component<Props, State> {
             tagged_message,
             getAvatarUrl,
             showTip = true,
+            renderAvatar,
```

Comment:
It would be helpful to understand how this is intended to be used. Feel free to reach out in the Preview team channel if it's more convenient.

Reply:
It would be helpful to have a screenshot of the intended experience, if one is available. Why are we diverging from the other FTUX tooltips, like the one in `SidebarNavSign`?

Reply:
> It would be helpful to have a screenshot of the intended experience, if one is available.

@megansmith-box [asked](https://github.com/box/box-ui-elements/pull/2880#discussion_r859109757) for the same thing. Isn't that a bit out of context of this change? I mean: This is a general purpose repository. Even if I'm extending a component here with a property I'll use in EndUserApp, the change in this repo should be examined in a more general context than that of my change in EUA. I have the screenshot of how the tooltip displayed on `CommentForm`'s `Avatar` looks like in EndUserApp's PR, which is PR that implements this tooltip. I could possibly link it here, but I was told to not post any links to our internal tools (Jira, GHE etc.) in this public repo. Is that true?

> Why are we diverging from the other FTUX tooltips, like the one in SidebarNavSign?

That's a good question! I consciously diverge from this pattern, as I think the old pattern does not make a good separation of concerns. `box-ui-elements` is a general purpose library exposing reusable UI components. They should be possible to be reused in any repository, or even in a context outside of Box organization (Am I wrong on that point?). Having concrete implementations of Box tooltips inside this repository does not follow that. First those tooltips are not in any way reusable - there are tied to some unique experiences for users in `EndUserApp` (or `AdminConsole` or some other parent repo). Second they are totally irrelevant for anyone outside of Box organisation.

So instead of defining my tooltip in `box-ui-elements` I decided it's better to define it in `EndUserApp` and pass it as a [Render Prop](https://reactjs.org/docs/render-props.html) to the component we need to display them on (wrap that component in my tooltip). That way `box-ui-elements` doesn't need to be aware of what exactly is done to the `Avatar`, how the tooltip is constructed nor what our user experiences are. All this logic will stay in `EndUserApp`, which is where I think it belongs much better.

Another reason is that this particular Persistent Onboarding tooltip I'm putting on `Avatar` is just one of the few tooltips in a bigger experience. 3 of those tooltips are applied on elements declared in `EndUserApp`, one on an element in `box-ui-elements` and one on an element in `box-content-preview`. I think's it's great to have the code for the whole experience defined in a single place, which also makes code reuse easy/possible. Following the pattern I'm divering from - declaring all tooltip logic in `box-ui-elements` (which would also mean declaring one tooltip in `box-content-preview) - would split a tightly related code in 3 repositories and would make code reuse very hard or impossible.

I'd say that's the pattern we should use in the future to declare tooltips on elements hidden somewhere deep in `box-ui-elements` - to pass render props and leave the tooltip declarations in parent repo (`EndUserApp`, `AdminConsole` etc.).

@jstoffan @ConradJChan @megansmith-box What do you think?